File: Cpp/1276_Number_of_Burgers_with_No_Waste_of_Ingredients.cpp
class Solution {
public:
    vector<int> numOfBurgers(int tomatoSlices, int cheeseSlices) {
        int x = (tomatoSlices - 2 * cheeseSlices) / 2;
        int y = (-tomatoSlices + 4 * cheeseSlices) / 2;
        if(x >= 0 && y >= 0 && (4*x + 2*y) == tomatoSlices && (x + y == cheeseSlices)){
            return {x, y};
        }
        
        return {};
    }
};

File: Kotlin/1276_Number_of_Burgers_with_No_Waste_of_Ingredients.kt
class Solution {
    fun numOfBurgers(tomatoSlices: Int, cheeseSlices: Int): List<Int> {
        val x = (tomatoSlices - 2 * cheeseSlices) / 2
        val y = (-tomatoSlices + 4 * cheeseSlices) / 2
        
        if(x >= 0 && y >= 0 && (4 * x + 2 * y) == tomatoSlices && (x + y == cheeseSlices)){
            return mutableListOf(x, y)
        }
        
        return mutableListOf()
    }
}